src/device-registry/utils/create-event.js

```diff
@@ -530,7 +513,26 @@ const createEvent = {
         data[0].data = !isEmpty(missingDataMessage) ? [] : data[0].data;
 
         logText("Setting cache...");
-        await createEvent.setCache(data, request);
+
+        try {
+          const resultOfCacheOperation = await Promise.race([
+            createEvent.setCache(data, request),
```

Comment:
properly handle cases where the Redis server is off, just to ensure that the entire function does not hang. @BenjaminSsempala